api/legacy_special_missions.php

```diff
@@ -6,13 +6,15 @@
 require "../classes/_autoload.php";
 
 $system = new System();
+$system->startTransaction();
 $system->is_api_request = true;
 
 try {
     $player = Auth::getUserFromSession($system);
     $player->loadData();
 } catch(Exception $e) {
     echo json_encode(['logout' => true]);
+    $system->rollbackTransaction();
```

Comment:
```suggestion
    $system->rollbackTransaction();
    error_log($e->getMessage());
```

We should capture these errors in server logs so we can see them